plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/MachineParameters.hs

```diff
@@ -0,0 +1,52 @@
+{-# LANGUAGE TypeFamilies  #-}
+{-# LANGUAGE TypeOperators #-}
+
+module PlutusCore.Evaluation.Machine.MachineParameters
+where
+
+import           PlutusCore.Constant
+
+import           PlutusCore.Core.Type                           hiding (Type)
+import           PlutusCore.Evaluation.Machine.BuiltinCostModel
+import           PlutusCore.Evaluation.Machine.ExBudget         ()
+
+import           GHC.Types                                      (Type)
+
+{-| We need to account for the costs of evaluator steps and also built-in function
+   evaluation.  The models for these have different structures and are used in
+   different parts of the code, so inside the valuator we pass separate objects
+   about most of the time .  It's convenient for clients of the evaluator to
+   only have to worry about a single object, so the CostModel type bundles the
+   two together.  We could conceivably have different evaluators with different
+   internal costs, so we keep the machine costs abstract.  The model for Cek
+   machine steps is in UntypedPlutusCore.Evaluation.Machine.Cek.CekMachineCosts.
+-}
+data CostModel machinecosts =
+    CostModel {
+      machineCostModel :: machinecosts
+    , builtinCostModel :: BuiltinCostModel
+    }
+
+{-| At execution time we need a 'BuiltinsRuntime' object which includes both the
+  cost model for builtins and their denotations.  This bundles one of those
+  together with the cost model for evaluator steps.  The 'term' type will be
+  CekValue when we're using this with the CEK machine. -}
+data MachineParameters machinecosts (term :: (Type -> Type) -> Type -> Type) (uni :: Type -> Type) (fun :: Type) =
```

Comment:
Maybe we need some `UNPACK` pragmas or something? the previous version passed these in as separate arguments, perhaps that matters?

Reply:
I think it was like that in an earlier, faster commit.  What I've done since then was to do the ledger interface stuff (which shouldn't have any effect) and rearranged the exports/imports, like importing things via `PlutusCore` instead of directly from `ExBudgetingDefaults`.